pkg/ccl/changefeedccl/kvfeed/kv_feed.go

```diff
@@ -749,10 +749,12 @@ func copyFromSourceToDestUntilTableEvent(
 		// spanFrontier returns the frontier timestamp for the specified span by
 		// finding the minimum timestamp of its subspans in the frontier.
 		spanFrontier = func(sp roachpb.Span) (sf hlc.Timestamp) {
+			first := true
```

Comment:
Instead of using a flag, I'd prefer to see something like:
```
		spanFrontier = func(sp roachpb.Span) (sf hlc.Timestamp) {
			sf = hlc.MaxTimestamp
			frontier.SpanEntries(sp, func(_ roachpb.Span, ts hlc.Timestamp) (done span.OpResult) {
				if ts.Less(sf) {
					sf = ts
				}
				return span.ContinueMatch
			})
			if sf == hlc.MaxTimestamp {
				sf = hlc.Timestamp{}
			}
			return sf
		}
```

Reply:
Or alternatively:
```
		spanFrontier = func(sp roachpb.Span) (sf hlc.Timestamp) {
			frontier.SpanEntries(sp, func(_ roachpb.Span, ts hlc.Timestamp) (done span.OpResult) {
				if ts.Less(sf) {
					sf = ts
				}
				if ts.IsEmpty() {
					return span.StopMatch
				}
				return span.ContinueMatch
			})
			return sf
		}
```

Reply:
I went with the first option but wasn’t entirely sure what the best way to handle 
```
if sf == hlc.MaxTimestamp {
	sf = hlc.Timestamp{}
}
```
Should we skip the event if the number of span entries here is zero (is that possible)? And whether it is possible for all subspans to have a timestamp of hlc.MaxTimestamp, in which case the minimum timestamp should be the `hlc.MaxTimestamp`. wdyt?